### DISCUSSION TITLE
Performance between CSS :hover and useHover (JS)

### DISCUSSION THREAD

**Discussion by boompttn** [adobe/react-spectrum#5669] [Q&A]
Is there a difference in performance between :hover selector (CSS) and useHover (JS)? If so, is the performance difference noticeable? If there's a big difference, will there be any alternatives since I only use it for styling

**Comment by boompttn** [comment]
It seems that whenever I hover enter and leave, the component re-renders but does not if using :hover selector

**Comment by reidbarber** [comment]
Check out this blog post: [Building a Button Part 2: Hover Interactions](https://react-spectrum.adobe.com/blog/building-a-button-part-2.html).

I would suggest looking at the nature of your application and considering whether the re-renders on hover will have a real-world impact on your application's performance.
